### ISSUE TITLE
Setting kbd_code_kana_host_key doesn't free the RALT key, which may be AltGr

### ISSUE THREAD

**Issue by MBilderbeek** [openMSX/openMSX#1056]
If you use "set kbd_code_kana_host_key" to move code/kana away from RALT (AltGr in some keyboards, where this setting is most useful for) it will map to the key you define, but it will not map away from RALT... This means that AltGr is still not free to use, whilst that was the whole purpose of the setting.

If you do this without an AltGr keyboard, you can also reproduce this. Although for those keyboards there isn't a big reason to do this, it still is strange.

Steps to reproduce:
1) start up with a turboR
2) set kbd_code_kana_host_key END
3) press RALT
4) (unexpected) see KANA LED toggle

The root cause is the following code in Keyboard.cc:

```
                if (unicode == 0) {
                        // It was an ambiguous key (numeric key-pad, CTRL+character)
                        // or a special key according to SDL (like HOME, INSERT, etc)
                        // or a first keystroke of a composed key
                        // (e.g. altr-gr + = on azerty keyboard) or driver is in
                        // direct SDL mapping mode:
                        // Perform direct SDL matrix to MSX matrix mapping
                        // But only when it is not a first keystroke of a
                        // composed key
                        if ((keyCode & Keys::KM_MODE) == 0) {
                                processSdlKey(time, down, key);
                        }
                }
```

NYYRIKKI reported this problem, and he is using a keyboard with AltGr. Apparently, in his case, there is no compose keystroke detected.

Anyway, in all cases what happens is that the RALT is mapped as matrix-to-matrix mode and in that mode it is directly mapped to the CODE/KANA key.

So, I think 2 issues:
1) if this setting is not on the default, it shouldn't use the hardcoded matrix-to-matrix mapping for RALT. (But how to implement that?)
2) Composed key detection is apparently failing. For this, I asked NYYRIKKI to give the output of 'trace_keypresses'.

**Comment by MBilderbeek** [comment]
NYYRIKKI wrote this on my request:

> I start to get a feeling that we've been here before... This output looks a bit funny, but indeed AltGr == CTRL+ALT... I anyway hope they can be separated from each other...
> 
> Key pressed, unicode: 0x0000, keyCode: 0x00132, keyName: LCTRL
> Key pressed, unicode: 0x0000, keyCode: 0x40133, keyName: RALT+CTRL
> Key released, unicode: 0x0000, keyCode: 0x480132, keyName: LCTRL+ALT,RELEASE
> Key released, unicode: 0x0000, keyCode: 0x400133, keyName: RALT,RELEASE

At least it explains why AltGr + G gives a BEEP in basic for NYYRIKKI. But it doesn't explain to me what this story is with the composed key trigger.

**Comment by MBilderbeek** [comment]
Some background on the AltGr == CTRL+ALT: https://bugzilla.mozilla.org/show_bug.cgi?id=900750
Especially:

> 
> Background: how this happens.
> 
> The Windows keyboard driver may define a flag KLLF_ALTGR. If present, the keyboard subsystem will emulate a LCTRL press/release around the RightAlt (VK_RMENU) press/release.  So, in presence of this flag, when physical AltGr is pressed, what an application sees is that LCTRL and RightAlt are pressed.
> 
> This flag is automatically enabled on all keyboards produced by MSKLC's (*THE* producer of Windows keyboard drivers; probably 99.9% of keyboards in use are produced by MSKLC) GUI when AltGr bindings are present (this may be avoided when using MSLKC's CLI).  So for most keyboards, if a character is produced when CTRL+ALT are active, it is produced when LCTRL and RightAlt are (logically) down.
> 
> As explained in https://bugzilla.mozilla.org/show_bug.cgi?id=880971, there is no way to predict which character is going to be produced based only on CapsLock/Shift/Alt/Control model.  One must know exactly ALL the keys which are now (logically) down. So when one investigates CTRL+ALT, one should choose which keys are down; one must use some heuristics.
> 
> It is natural to choose the most frequent combination: LCTRL+RightAlt.

**Comment by MBilderbeek** [comment]
I think this was fixed in e906b33f7318503f14dda718f47c6ff847bf4038
